src/gluonts/dataset/split.py

```diff
@@ -306,8 +306,12 @@ def test_pair(
         self, entry: DataEntry, prediction_length: int, offset: int = 0
     ) -> Tuple[DataEntry, DataEntry]:
         offset_ = self.offset + offset
-        if self.offset < 0 and offset_ >= 0:
-            offset_ += len(entry)
+        if self.offset < 0:
+            offset_ += entry[FieldName.TARGET].shape[-1]
+        assert (
+            offset_ + prediction_length <= entry[FieldName.TARGET].shape[-1]
+        ), "Offset too short to generate windows"
```

Comment:
Also here "short" sounds strange. To give proper context maybe a message like this would work better

```
"Not enough data to generate some of the windows; try splitting data at an earlier offset"
"Not enough data to generate some of the windows; try splitting data at an earlier date"
```

Reply:
I agree with you. I adapted the assertions.